### PR TITLE
feat: number autocomplete

### PR DESCRIPTION
**Pull request by XiLaiTL**
I have made a demo about the complete the "123" to Chinese number “一百二十三”

However, it work only when you type a space before you enter the number.
And I just made the int and have no idea about creating the float for "0.123" to "一分二釐三毫" and for numbers larger than "十億"

I am sorry I don't have time to improve the code. If you have time, please improve it or continue to solve the number problem.
Thanks.

### CODE REVIEW

**Comment by antfu** [comment]
Can you use `Wenyan.num2hanzi` to do that. We already implemented this feature in the online ide where you can take it as a reference

https://github.com/wenyan-lang/ide/pull/1

https://github.com/wenyan-lang/ide/blob/9bb7487c27ae66d7b649c2365524e3b66fea90a8/static/assets/js/ide.js#L740-L746

**Comment by XiLaiTL** [comment]
Thanks! This problem is solved and sorry I am a green hand of TS that I don't know how to import the module of Wenyan so in the file I copy the './converts/hanzi2num' when debugging in my VS Code. 
So it will be a new problem. sorry!!

File: src/editor/dynamicSnippets.ts [hunk 1]
```diff
@@ -62,7 +63,36 @@ function onTextChanged(e: TextDocumentChangeEvent) {
   }
 }
 
-const m: ExtensionModule = () => {
+function provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken, context: CompletionContext) {
+  const line = document.lineAt(position)
+  const lineText = line.text.substring(0, position.character)
+  let lenth = lineText.length
+  for (lenth; lenth >= 0; lenth -= 1) {
+    if (lineText[lenth - 1] !== '.' && lineText[lenth - 1] !== '0' && lineText[lenth - 1] !== '1' && lineText[lenth - 1] !== '2' && lineText[lenth - 1] !== '3' && lineText[lenth - 1] !== '4' && lineText[lenth - 1] !== '5' && lineText[lenth - 1] !== '6' && lineText[lenth - 1] !== '7' && lineText[lenth - 1] !== '8' && lineText[lenth - 1] !== '9')
```
Comment (antfu) [inline src/editor/dynamicSnippets.ts:71]:
```suggestion
    if (!lineText[lenth - 1].match(/[0-9.]/)
```

**Comment by XiLaiTL** [comment]
Ok, applying regular expression solve the problem of matching when number is after the name.
Thanks!

**Comment by antfu** [comment]
I have just invited you to be the collaborator of this project, Thank you!

**Comment by XiLaiTL** [comment]
Thanks！
But there is still some bug in the newest version: the number is still in 123.

**Comment by antfu** [comment]
I would be grateful if you can help to look into it, I don't have much bandwidth to maintain this project, and if you want, I can set up the workflow so you can maintain it and do the releases.

**Comment by XiLaiTL** [comment]
It seems that the @wenyan/core can't use in the plugins, as I test the num2hanzi in console.log and it returns none.
A possible solution is adding the source file of 'wenyan/src/converts/hanzi2num.ts in this package.
Fine, if I can help you.

**Pull request merged**